### ISSUE TITLE
Move smoke test into standalone script that can be run locally

### ISSUE THREAD

**Issue by jcbhmr** [devcontainers-community/templates#3]
It would be nice to be able to run tests locally to make sure everything "kinda works". Right now I don't _think_ this is possible.

- [ ] Create some kind of `tools/test.sh` file
- [ ] Move logic from https://github.com/devcontainers-community/templates/tree/main/.github/actions/smoke-test to there
- [ ] Make sure it still works
- [ ] Add a GitHub Action to test changes

**Comment by eitsupi** [comment]
Hmmm...wouldn't this be better implemented upstream?

**Comment by jcbhmr** [comment]
Good idea! 👍 I think doing a PoC or something in this repo might be a good idea in parallel to a PR to the official template or something? 🤷‍♂️

You're right though about this being a good thing to bring up with the upstream repo

**Comment by eitsupi** [comment]
It would definitely be nice to have an easy way to try out the templates. (See also my comment in rocker-org/devcontainer-templates#1)

It's great that you are working on it too, but I just thought it would be better to discuss it with devcontainers before devoting your valuable time to this.
They get paid full time to work on devcontainers and are very knowledgeable.

**Comment by jcbhmr** [comment]
> They get paid full time to work on devcontainers and are very knowledgeable.

Legit good point I forgot about 😅

Good idea to ask them first then 👍

**Comment by jcbhmr** [comment]
Open an issue with template-starter https://github.com/devcontainers/template-starter/issues/15

**Comment by jcbhmr** [comment]
> Thanks for combining it into a single file! I'll add a ./test.sh or tools/test.sh or something to my own repos to just invoke that .github/actions/smoke-test/test.sh script.
